mindsdb/integrations/handlers/leonardoai_handler/leonardo_ai_handler.py

```diff
@@ -0,0 +1,234 @@
+            4. leonardo.api_key setting in config.json
+        """
+
+        # 1
+        if "api_key" in args["using"]:
+            return args["using"]["api_key"]
+        # 2
+        connection_args = engine_storage.get_connection_args()
+        if "api_key" in connection_args:
+            return connection_args["api_key"]
+        # 3
+        api_key = os.getenv("LEONARDO_API_KEY")
+        if api_key is not None:
+            return api_key
+        # 4
+        config = Config()
+        leonardo_cfg = config.get("leonardo", {})
+        if "api_key" in leonardo_cfg:
+            return leonardo_cfg["api_key"]
+
+        if strict:
+            raise Exception(
+                'Missing API key "api_key". Either re-create this ML_ENGINE specifying the `api_key` parameter,\
+                 or re-create this model and pass the API key with `USING` syntax.'
+            )
+
+    def _get_platform_model(self, args):
+        """
+        Returns a list of available model based on the API key provided
+        """
+        model_ids = []
+
+        args = self.model_storage.json_get('args')
+        api_key = self._get_leonardo_api_key(args, self.engine_storage)
+
+        self.connection = requests.get(
+            "https://cloud.leonardo.ai/api/rest/v1/platformModels",
+            headers={
+                "accept": "application/json",
+                "authorization": f"Bearer {api_key}"
+            }
+        )
+
+        models = self.connection.json()
+
+        # extract the model ids from the response
+        model_ids = [model['id'] for model in models['custom_models']]
+
+        return model_ids
+
+    def predict_answer(self, prompts, **kwargs):
+        """
+        Generates pictures based on the prompts and returns URLs with few variations.
+
+        Request Flow:
+            - POST request with a prompt is sent
+            - `generation_id` is created for the request
+            - POST request will take couple of seconds to generate the picture, till then the process will be kept busy with a simple math calculation.
+            - New GET request with the `generation_id` will fetch the generated pictures as URLs
+        """
+        args = self.model_storage.json_get('args')
+        api_key = self._get_leonardo_api_key(args, self.engine_storage)  # fetch API key
+        generation_id = ''
+
+        # Endpoint URL
+        generation_url = "https://cloud.leonardo.ai/api/rest/v1/generations"
+
+        post_headers = {
+            "accept": "application/json",
+            "content-type": "application/json",
+            "authorization": f"Bearer {api_key}"
+        }
+
+        get_headers = {
+            "accept": "application/json",
+            "authorization": f"Bearer {api_key}"
+        }
+
+        # payload
+        generation_payload = {
+            "height": 512,
+            "modelId": args['using']['model'],
+            "prompt": f"{prompts}",
+            "width": 512
```

Comment:
Should we add width and height as params in USING clause so users can specify them?

Reply:
User can now pass height and width, if user doesn't provide it, default values will be used.